src/log_file.mli

```diff
@@ -0,0 +1,63 @@
+(*
+ * Copyright (c) 2021 Tarides <contact@tarides.com>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *)
+
+open! Import
+
+module Make (IO : Io.S) (Key : Data.Key) (Value : Data.Value) : sig
+  module Entry : Data.Entry.S with type key := Key.t and type value := Value.t
+
+  type t
+  type key := Key.t
+  type value := Value.t
+
+  val create : IO.t -> t
```

Comment:
maybe add a comment here that it also syncs the in-memory data with the disk? since we used to do this in two  steps before (the same for `replace`)